src/app/setup-wizard/data-collection/on-site-assessment/assessment-details-form/assessment-details-form.component.ts

```diff
@@ -60,6 +66,25 @@ export class AssessmentDetailsFormComponent {
     this.energyEquipmentSub.unsubscribe();
   }
 
+  setUtilityTypes() {
```

Comment:
Both of these functions should be tagged `async` and add `await` to the saveChanges calls

`await this.saveChanges()`